### PR TITLE
Add blogpost about puppet4 migration

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+We announced the deprecation of Puppet 3 in our modules
+[already](https://voxpupuli.org/blog/2016/12/22/putting-down-puppet-3/). There
+are a few steps needed to be taken before a Puppet-4-only release can be done:
+
+* Create a new branch which is called `puppet`
+* Check the current metadata.json for the required Puppet version and bump it if needed
+  * We already have a few modules that only support Puppet 4
+  * The minimal required Puppet Version should not be 4.0.0 but [4.6.1](https://github.com/voxpupuli/community-triage/blob/master/modules/notes/2017-01-05.md#discussion)
+* Check the [.sync.yml](https://github.com/voxpupuli/puppet-tea/blob/e49d6d1ce8ba71c2123edf9fae45cde19e603ec3/.sync.yml#L3-L17), sometimes the `.travis.yml` or `Gemfile` files are unmanaged or modified because the module is already Puppet 4 only. Remove the settings from `.sync.yml`
+* Do a modulesync with at least version XY ([how to do it](https://github.com/voxpupuli/modulesync_config#how-to-use-it))
```
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: xy: The needed changes are already merged in modulesync_config, but unreleased.  https://github.com/voxpupuli/modulesync_config/pull/297

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+github_username: bastelfreak
+twitter_username: BastelsBlog
+---
+
+
+We announced the deprecation of Puppet 3 in our modules
+[already](https://voxpupuli.org/blog/2016/12/22/putting-down-puppet-3/). There
+are a few steps needed to be taken before a Puppet-4-only release can be done:
+
+* Create a new branch which is called `puppet`
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: This should be `puppet3`

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+* Check the current metadata.json for the required Puppet version and bump it if needed
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: This step should come first, as there would be no need for a `puppet3` branch for modules that do not currently support it.

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  * We already have a few modules that only support Puppet 4
+  * The minimal required Puppet Version should not be 4.0.0 but [4.6.1](https://github.com/voxpupuli/community-triage/blob/master/modules/notes/2017-01-05.md#discussion)
+* Check the [.sync.yml](https://github.com/voxpupuli/puppet-tea/blob/e49d6d1ce8ba71c2123edf9fae45cde19e603ec3/.sync.yml#L3-L17), sometimes the `.travis.yml` or `Gemfile` files are unmanaged or modified because the module is already Puppet 4 only. Remove the settings from `.sync.yml`
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: Can we provide an example of what to look for here? A "good" vs "bad" would ensure some clarity.

**Comment by rnelson0** [comment]
While this blog post is for Vox Pupuli, I think the pattern may be followed by others, so it will pay to be a little more verbose in what to look for and change.

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 2 above)
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: I think we also want to state that a final puppet 3 release, announcing the deprecation of puppet 3 support in future versions, should be released prior to creating the branch.

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 3 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: fixed

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 2 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: fixed

**Comment by bastelfreak** [comment]
@rnelson0 I made some updates

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+
+We announced the deprecation of Puppet 3 in our modules
+[already](https://voxpupuli.org/blog/2016/12/22/putting-down-puppet-3/). There
+are a few steps needed to be taken before a Puppet-4-only release can be done:
+
+* Check the current metadata.json for the required Puppet version and bump it if needed
+  * We already have a few modules that only support Puppet 4, in this case you can ignore this guide
+  * The minimal required Puppet Version should not be 4.0.0 but [4.6.1](https://github.com/voxpupuli/community-triage/blob/master/modules/notes/2017-01-05.md#discussion)
+  * Also check if the dependencies in the `metadata,json` all have Puppet 4 support, probably they have to be bumped too
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: `metadata.json`

Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: change `probably they have` to `they may have`, it flows a little better.

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  * The minimal required stdlib version has to be 4.6.0
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: `minimum`

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+* Do a last release with Puppet 3 support which announces the next major release without Puppet 3
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: How about `Perform a release of the module within the current Major version that announces the deprecation of Puppet 3. e.g. Current version is 2.1.3, release 2.2.0 with the deprecation notice`

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+* Create a new branch which is called `puppet3`
+  * The required Puppet version for this branch in the `metadata.json` should be the latest available, which is currently 3.8.7
+  * Keep in mind that stdlib [4.13.0](https://forge.puppet.com/puppetlabs/stdlib/changelog#supported-release-4130) deprecates a lot of functions, you maybe want to require an older version in the `puppet3` branch
+* Check the [.sync.yml](https://github.com/voxpupuli/puppet-tea/blob/e49d6d1ce8ba71c2123edf9fae45cde19e603ec3/.sync.yml#L3-L17), sometimes the `.travis.yml` file is unmanaged or modified because the module is already Puppet 4 only. Take a look for an `include` block in the `.travis.yml` section and remove it
+* Do a modulesync with at least version XY ([how to do it](https://github.com/voxpupuli/modulesync_config#how-to-use-it))
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: I assume we need XY to be filled in before merging, right?

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+* Merge any incoming pull requests that add Puppet 4 functionality now, but not earlier
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: `You may now merge any existing Pull Requests to the master branch that are not backwards compatible with Puppet 3` ?

File: _posts/2017-01-09-migrating-to-puppet4.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+* Do a major version bump + a release soon after the first Puppet 4 functionalty got added
```
Comment (rnelson0) [inline _posts/2017-01-09-migrating-to-puppet4.md]: `functionality is`

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 5 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: fixed

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 6 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: fixed

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 10 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: fixed

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 8 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: fixed

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 7 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: okay

File: _posts/2017-01-09-migrating-to-puppet4.md (hunk 9 above)
Comment (bastelfreak) [inline _posts/2017-01-09-migrating-to-puppet4.md]: okay

**Pull request merged**